app/models/candidate_interface/work_experience_form.rb

```diff
@@ -3,15 +3,25 @@ class WorkExperienceForm
     include ActiveModel::Model
 
     attr_accessor :role, :organisation, :details, :working_with_children,
-                  :commitment, :start_date_month, :start_date_year,
-                  :end_date_month, :end_date_year
+                  :commitment,
+                  :start_date_day, :start_date_month, :start_date_year,
+                  :end_date_day, :end_date_month, :end_date_year
 
     validates :role, :organisation, :details, :working_with_children,
-              :commitment, :start_date_month, :start_date_year,
+              :commitment,
               presence: true
 
     validates :working_with_children, inclusion: { in: %w(true false) }
 
+    validate :start_date_valid
+    validate :start_date_after_end_date
+
+    validates :role, :organisation,
+              length: { maximum: 60 }
+
+    validates :details,
+              word_count: { maximum: 150 }
+
     def save(application_form)
```

Comment:
Non-blocking for this PR as it's about validations but something I noticed: Any reason why we don't have specs for `#save`? It also behaves differently to our other form object models as it returns the object instead of `true` or `false`. Is this okay?

Reply:
Sorted ✅